### ISSUE TITLE
Incorrect subsitution of partial derivatives by .subs() 

### ISSUE THREAD

**Issue by nschaeff** [sympy/sympy#9135]
With sympy 0.7.6, the subs() method does a wrong substitution:

    from sympy import *
    init_printing()

    x,y,z=symbols('x,y,z')
    f=Function('f')

    expr = diff(f(x,y),x,x,y)
    pprint(expr)

    e2 = expr.subs(diff(f(x,y),y,2), z)
    pprint(e2)

The call to subs() method incorrectly substitutes a `df/dy` with `z`, while we asked to substitute `d2f/dy2` instead !

**Comment by pbrady** [comment]
Same behavior on 0.7.4 & 0.7.5

**Comment by asmeurer** [comment]
I don't understand what the derivative algorithm was doing. It was clearly wrong. None of the tests were testing cases that shouldn't be replaced. I rewrote it using a simple multiset check. https://github.com/sympy/sympy/pull/10014

**Comment by wulu473** [comment]
I think there might still be a problem. Version 1.0 gives me
```
>>> from sympy import *
>>> f,g,x,y = symbols('f g x y')
>>> diff(f(x,y),x,x).subs(diff(f(x,y),y),g)
Derivative(g, x, x)
```

**Comment by bjodah** [comment]
@wulu473 this appears to be fixed in the release candidate for v1.1
